pkg/globalmanager/controllers/federatedlearning/federatedlearningjob.go

```diff
@@ -402,21 +402,128 @@ func IsJobFinished(j *sednav1.FederatedLearningJob) bool {
 	return false
 }
 
+func (c *Controller) getSecret(namespace, name, ownerStr string) (secret *v1.Secret, err error) {
+	if name != "" {
+		secret, err = c.kubeClient.CoreV1().Secrets(namespace).Get(context.TODO(), name, metav1.GetOptions{})
+		if err != nil {
+			err = fmt.Errorf("failed to get the secret %s for %s: %w",
+				name,
+				ownerStr, err)
+		}
+	}
+
+	return
+}
+
+func (c *Controller) getModel(ctx context.Context, namespace, name string) (model *sednav1.Model, secret *v1.Secret, err error) {
+	if name != "" {
+		model, err = c.client.Models(namespace).Get(ctx, name, metav1.GetOptions{})
+		if err != nil {
+			err = fmt.Errorf("failed to get the model %s: %w", name, err)
+		}
+	}
+
+	if model != nil {
+		secret, err = c.getSecret(
+			namespace,
+			model.Spec.CredentialName,
+			fmt.Sprintf("model %s", name),
+		)
+	}
+
+	return
+}
+
+func (c *Controller) getDataset(ctx context.Context, namespace, name string) (dataset *sednav1.Dataset, secret *v1.Secret, err error) {
+	if name != "" {
+		dataset, err = c.client.Datasets(namespace).Get(ctx, name, metav1.GetOptions{})
+		if err != nil {
+			err = fmt.Errorf("failed to get the dataset %s: %w", name, err)
+		}
+	}
+
+	if dataset != nil {
+		secret, err = c.getSecret(
+			namespace,
+			dataset.Spec.CredentialName,
+			fmt.Sprintf("model %s", name),
+		)
+	}
+
+	return
+}
+
+// addCRToWorkerMounts adds CR(e.g., model, dataset) to worker mounts.
+func (c *Controller) addCRToWorkerMounts(workerMount []runtime.WorkerMount, model *sednav1.Model, dataset *sednav1.Dataset,
+	secret *v1.Secret, downloadByInitializer bool, envName string) ([]runtime.WorkerMount, bool) {
+	var url string
+	if model != nil {
+		url = model.Spec.URL
+	} else if dataset != nil {
+		url = dataset.Spec.URL
+	}
+
+	if url != "" {
+		workerMount = append(workerMount,
+			runtime.WorkerMount{
+				URL: &runtime.MountURL{
+					URL:                   url,
+					Secret:                secret,
+					DownloadByInitializer: downloadByInitializer,
+				},
+				EnvName: envName,
+			},
+		)
+		return workerMount, true
+	}
+
+	return nil, false
+}
+
+// addTransmitterToWorkerParam adds transmitter to the WorkerParam
+func (c *Controller) addTransmitterToWorkerParam(param *runtime.WorkerParam, job *sednav1.FederatedLearningJob) error {
+	transmitter := job.Spec.Transmitter
+
+	if transmitter.S3Transmitter.AggregationDataPath != "" {
```

Comment:
use nil to indicate whether using s3 or not
```suggestion
	if transmitter.S3Transmitter != nil {
```

Reply:
ok, fix it